src/Services/AbstractQueuedJob.php

```diff
@@ -271,4 +271,31 @@ public function __get($name)
     {
         return isset($this->jobData->$name) ? $this->jobData->$name : null;
     }
+
+    /**
+     * Resolves a queue name to one of the queue constants.
+     * If $queue is already the value of one of the constants, it will be returned.
+     * If the queue is unknown, `null` will be returned.
+     */
+    public static function getQueue(string|int $queue): ?string
+    {
+        switch (strtolower($queue)) {
+            case 'immediate':
+                $queue = QueuedJob::IMMEDIATE;
+                break;
+            case 'queued':
+                $queue = QueuedJob::QUEUED;
+                break;
+            case 'large':
+                $queue = QueuedJob::LARGE;
+                break;
+            default:
+                $queue = (string) $queue;
+                $queues = [QueuedJob::IMMEDIATE, QueuedJob::QUEUED, QueuedJob::LARGE];
+                if (!ctype_digit($queue) || !in_array($queue, $queues)) {
+                    return null;
+                }
+        }
+        return $queue;
```

Comment:
```suggestion
        if (is_int($queue)) {
            return $queue;
        }
        return match(strtolower($queue)) {
            'immediate' => QueuedJob::IMMEDIATE,
            'queued' => QueuedJob::QUEUED,
            'large' => QueuedJob::LARGE,
            default => null,
        }
```

Not sure of the value of having string|int as a param is, string only would be better.

Method name might benefit from being called something like getQueueFromType

Reply:
`int` is there because the consts are string values of ints (`"0"`, `"1"`, etc. So it makes sense to allow someone to pass in `1` and get the corresponding `"1"` for `===` matching against the const.

Reply:
I'm not sure how `getQueueFromType` is better.... maybe `parseValueToQueueConst` or something like that, if you really want it renamed?

Reply:
All good, I'm not too worried about the rename given it's the same name as the old method